src/movies.js

```diff
@@ -1,22 +1,122 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+
+    let directors = moviesArray.map(function(movies) {
+        return movies.director;
+    });
+
+    return directors;
+    
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+
+    let spielberg = moviesArray.filter((director) => director.director === 'Steven Spielberg')
+    let howMany = spielberg.filter((drama) => drama.genre.includes('Drama')) 
+    
+    return howMany.length
+
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (moviesArray.length === 0) return 0;
+    let scores = moviesArray.map(function(score) {
+
+        return score.score;
+     
+    })
+
+    let sumScores = scores.reduce((acc, val) =>{
+        
+        if(!val) return acc + 0
+        else return acc + val
+    } , 0)
+    let average = +(sumScores / scores.length).toFixed(2);
+    return  average;
+
+    
+    
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
```

Comment:
Instead of copying the logic of the scoresAverage function, you could've returned the scoresAverage function itself here.
Like: `return scoresAverage(numDramas);`